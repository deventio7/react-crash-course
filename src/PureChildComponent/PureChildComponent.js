import React from 'react';
import PropTypes from 'prop-types';

const PureChildComponent = ({ passedText }) => {
  console.log('Rendering PureChildComponent');
  return (
    <div>
      PureChildComponent<br/>
      THIS COMPONENT RECEIVED: '{ passedText }'
    </div>
  );
}

PureChildComponent.propTypes = {
  passedText: PropTypes.string.isRequired
}

export default PureChildComponent;