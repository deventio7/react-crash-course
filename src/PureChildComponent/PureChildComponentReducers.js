import SampleChildComponentActionTypes from '../SampleChildComponent/SampleChildComponentActionTypes';

const initialState = {
  passedText: 'initial text'
}

const sampleChildSubmit = (state = initialState, action) => {
  switch (action.type) {
    case SampleChildComponentActionTypes.SAMPLE_CHILD_COMPONENT_SUBMIT:
      return {
        ...state,
        passedText: action.payload
      };
    default:
      return state;
  }
}

export default {
  sampleChildSubmit
};