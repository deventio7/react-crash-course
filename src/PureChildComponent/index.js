import PureChildComponent from './PureChildComponent';

export default PureChildComponent;