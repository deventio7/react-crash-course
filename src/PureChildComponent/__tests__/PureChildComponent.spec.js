import React from 'react';
import { shallow } from 'enzyme';
import PureChildComponent from '../PureChildComponent';

describe('PureChildComponent', () => {
  it('renders and matches snapshot', () => {
    const wrapper = shallow(<PureChildComponent passedText="TEST TEXT"/>);
    expect(wrapper).toMatchSnapshot();
  });
})
