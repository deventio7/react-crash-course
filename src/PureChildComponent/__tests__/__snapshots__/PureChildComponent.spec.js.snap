// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PureChildComponent renders and matches snapshot 1`] = `
<div>
  PureChildComponent
  <br />
  THIS COMPONENT RECEIVED: '
  TEST TEXT
  '
</div>
`;
